### ISSUE TITLE
[BUG] matxException thrown

### ISSUE THREAD

**Issue by siegLoesch** [NVIDIA/MatX#432]
**Describe the bug**
During execution of example1_simple_slice an exception is thrown:
_Tensor{int32_t} Rank: 2, Sizes:[2, 2], Strides:[4,1]
terminate called after throwing an instance of 'matx::detail::matxException'
  what():  matxException (matxNotSupported: mtype == CU_MEMORYTYPE_HOST || mtype == 0: Invalid memory type for printing) - /opt/matx/include/matx/core/tensor_utils.h:822
Aborted_

**To Reproduce**
1a. Taking cmake with _[CMakeLists.zip](https://github.com/NVIDIA/MatX/files/11680430/CMakeLists.zip)_ to configure and build the cuda file:
```
#include <matx.h>

using namespace matx;

int main() {
	auto t2 = make_tensor<int>({5, 4});
	// Initialize the tensor linearly
	t2.SetVals({
		{1, 2, 3, 4},
		{5, 6, 7, 8},
		{9, 10, 11, 12},
		{13, 14, 15, 16},
		{17, 18, 19, 20}
	});
	// TODO: Create a slide of the view t2 starting at the second element and
	// ending at the third element (inclusive) in both dimensions
	auto t2s = slice(t2, {1,1}, {3,3});
	t2s.PrefetchDevice(0);
	print(t2s);
	return 0;
}
```

2a. Running the resulting executable _ex1SimpleSlice_ throws above exception.
To countercheck:
1b. Goto directory _MatX/docs_input/notebooks_ 
2b. Modify the shell script _compile_and_run.sh_ inside subdir _exercises_ as can be seen inside 
_[compile_and_run.zip](https://github.com/NVIDIA/MatX/files/11680496/compile_and_run.zip)_ 
(At least change 'ENABLE_CUTLASS' to 'MATX_ENABLE_CUTLASS', add -lnvToolsExt, add -lcuda and adapt the python version).
2b. Run _bash exercises/compile_and_run.sh example1_simple_slice_ whichs throws the same exception.

**Expected behavior**
A print of the sliced tensor.

**Code snippers**
See above attached files.

**System details (please complete the following information):**
 - OS: Debian 11, Bullseye
 - CUDA version: [e.g CUDA 11.8]
 - g++ version: [e.g. 10.2.1]

**Additional context**
As soon as the print commant inside the cuda source file is commented out the execution terminates without error.
Appreciate any help from your side to proceed with the training material - thank you!
Siegfried

**Comment by cliffburdick** [comment]
That's interesting -- which GPU is this? I'm not sure why this wouldn't be triggering in our code. I can submit a patch if I can't reproduce it

**Comment by siegLoesch** [comment]
The GPU is: GeForce RTX 2060
BR
Siegfried

**Comment by cliffburdick** [comment]
Hi @siegLoesch can you please try the `print_device` branch?

**Comment by siegLoesch** [comment]
Hello @cliffburdick,
that works well. Output from example1_simple_slice is (added original tensor output to check correctness of slice):
_Tensor{int32_t} Rank: 2, Sizes:[5, 4], Strides:[4,1]
000000: 1 2 3 4 
000001: 5 6 7 8 
000002: 9 10 11 12 
000003: 13 14 15 16 
000004: 17 18 19 20 
Tensor{int32_t} Rank: 2, Sizes:[2, 2], Strides:[4,1]
000000: 6 7 
000001: 10 11_
Tested also with _example1_init_ and _example1_permute_ which also yield correct results.

Thank you for your efforts and kind regards
Siegfried

**Comment by cliffburdick** [comment]
Resolved by #436
